core/internals.h

```diff
@@ -455,6 +455,9 @@ void observe_clear(lwm2m_context_t * contextP, lwm2m_uri_t * uriP);
 bool observe_handleNotify(lwm2m_context_t * contextP, void * fromSessionH, coap_packet_t * message, coap_packet_t * response);
 void observe_remove(lwm2m_observation_t * observationP);
 lwm2m_observed_t * observe_findByUri(lwm2m_context_t * contextP, lwm2m_uri_t * uriP);
+#ifndef LWM2M_VERSION_1_0
```

Comment:
It might be useful to add a check for `LWM2M_SERVER_MODE` in addition to `LWM2M_VERSION_1_0`.